### PR TITLE
Added email sending and templates

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/met-public/issues/<Put the github issue number here>

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/services/email_verification_service.py [hunk 1]
```diff
@@ -53,11 +62,30 @@ def verify(self, verification_token, survey_id):
...
-        """Send an verification email."""
-        print(email_verification)
-        # TODO: send email.
-        return DefaultMethodResult(True, '', 0)
+    def _send_verification_email(email_verification: dict) -> None:
+        """Send an verification email.Throws error if fails."""
+        sender = current_app.config.get('MAIL_FROM_ID')
+        survey_id = email_verification.get('survey_id')
+        email_to = email_verification.get('email_address')
+        survey: SurveyModel = SurveyModel.get_survey(survey_id)
```
Comment (jadmsaadaot) [inline met-api/src/met_api/services/email_verification_service.py:70]: I think better to use get_open_survey here to make sure it is also open for user.

File: met-api/src/met_api/services/email_verification_service.py [hunk 2]
```diff
@@ -53,11 +62,30 @@ def verify(self, verification_token, survey_id):
...
+        if not survey:
+            raise ValueError('Survey not found')
+        template = ENV.get_template('email_templates/email_verification.html')
+        # TODO make it read from config
+        subject = 'survey link - link expires in 24h'
```
Comment (jadmsaadaot) [inline met-api/src/met_api/services/email_verification_service.py:75]: we might need to discuss how to make this dynamic. If later a dev changes expiry date they might leave this here.

File: met-api/src/met_api/services/email_verification_service.py [hunk 3]
```diff
@@ -53,11 +62,30 @@ def verify(self, verification_token, survey_id):
...
+        app_url = f"{g.get('origin_url', '')}/" \
+                  f"{current_app.config.get('SURVEY_PATH')}" \
+                  f"{survey_id}/{email_verification.get('verification_token')}"
+        body = template.render(engagement_name=survey.get('engagement').get('name'), url=app_url)
```
Comment (jadmsaadaot) [inline met-api/src/met_api/services/email_verification_service.py:79]: what if engagement was None?

Comment (saravanpa-aot) [reply met-api/src/met_api/services/email_verification_service.py:81]: i added a check.its a private method ;ideally shud be just receiving what it needs ;but it was lot of refactoring ..will try to improve it..

File: met-api/email_templates/email_verification.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<h1>Thank you for your interest in sharing your thoughts about {{ engagement_name }}.</h1>
+
+Please click the link below to access the survey and share your comments. This link will expire in 24h.
+
+<a href='{{ url }}'>Access Survey</a>
+
+You can view the survey results anytime on . The comments will be available for viewing when the Engagement period is over, starting .
```
Comment (tiagograf-AOT) [inline met-api/email_templates/email_verification.html:7]: did you miss to add the date of engagement after the starting... ?

File: met-api/src/met_api/services/email_verification_service.py [hunk 5]
```diff
@@ -20,7 +29,7 @@ def get_active(self, verification_token):
         self.validate_object(db_email_verification)
         return db_email_verification
 
-    def create(self, email_verification: EmailVerificationSchema) -> DefaultMethodResult:
+    def create(self, email_verification: EmailVerificationSchema) -> None:
```
Comment (tiagograf-AOT) [inline met-api/src/met_api/services/email_verification_service.py:32]: Return type should be the email verification instead of None

**Review by tiagograf-AOT — APPROVED** [review]

**Review by djnunez-aot — APPROVED** [review]

**Review by jadmsaadaot — APPROVED** [review]

**Pull request merged**